packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -1101,6 +1102,17 @@ async function handler(
   const calEventUserFieldsResponses =
     "calEventUserFieldsResponses" in reqBody ? reqBody.calEventUserFieldsResponses : null;
 
+  const iCalUID = getICalUID({
+    event: { iCalUID: originalRescheduledBooking?.iCalUID, uid: originalRescheduledBooking?.uid },
+    uid,
+  });
+  // For bookings made before introducing iCalSequence, assume that the sequence should start at 1. For new bookings start at 0.
+  const iCalSequence = originalRescheduledBooking?.iCalSequence
+    ? originalRescheduledBooking.iCalSequence + 1
+    : originalRescheduledBooking
+    ? 1
+    : 0;
+
```

Comment:
nit: It deserves it's own function IMHO.

function getIcalSequence(originalRescheduledBooking) {
      // Fresh booking
	if (!originalRescheduleBooking) {
	   return 0;
	}
	// Reschedule but iCalSequence not available. So, it should atleast be 1, assume 1.
	if (!originalRescheduleBooking.iCalSequence) { 
	  return 1;
	}
	
	return originalRescheduleBooking.iCalSequence + 1;
}